src/Connectors/PhpRedisConnector.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace Monospice\LaravelRedisSentinel\Connectors;
+
+use Illuminate\Support\Arr;
+use Illuminate\Redis\Connectors\PhpRedisConnector as LaravelPhpRedisConnector;
+use LogicException;
+use Monospice\LaravelRedisSentinel\Connections\PhpRedisConnection;
+use Monospice\LaravelRedisSentinel\Exceptions\RedisRetryException;
+use Redis;
+use RedisSentinel;
+use RedisException;
+
+/**
+ * Initializes PhpRedis Client instances for Redis Sentinel connections
+ *
+ * @category Package
+ * @package  Monospice\LaravelRedisSentinel
+ * @author   Jeffrey Zant <j.zant@slash2.nl>
+ * @license  See LICENSE file
+ * @link     http://github.com/monospice/laravel-redis-sentinel-drivers
+ */
+class PhpRedisConnector extends LaravelPhpRedisConnector
+{
+    /**
+     * Holds the current sentinel servers.
+     *
+     * @var array
+     */
+    protected $servers;
+
+    /**
+     * The number of times the client attempts to retry a command when it fails
+     * to connect to a Redis instance behind Sentinel.
+     *
+     * @var int
+     */
+    protected $retryLimit = 20;
+
+    /**
+     * The time in milliseconds to wait before the client retries a failed
+     * command.
+     *
+     * @var int
+     */
+    protected $retryWait = 1000;
+
+    /**
+     * Configuration options specific to Sentinel connection operation
+     *
+     * Some of the Sentinel configuration options can be entered in this class.
+     * The retry_wait and retry_limit values are passed to the connection.
+     *
+     * @var array
+     */
+    protected $sentinelKeys = [
+        'sentinel_timeout' => null,
+        'retry_wait' => null,
+        'retry_limit' => null,
+        'update_sentinels' => null,
+
+        'sentinel_persistent' => null,
+        'sentinel_read_timeout' => null,
+    ];
+
+    /**
+     * Create a new Redis Sentinel connection from the provided configuration
+     * options
+     *
+     * @param array $server  The client configuration for the connection
+     * @param array $options The global client options shared by all Sentinel
+     * connections
+     *
+     * @return PhpRedisConnection The Sentinel connection containing a configured
+     * PhpRedis Client
+     */
+    public function connect(array $servers, array $options = [ ])
+    {
+        // Set the initial Sentinel servers.
+        $this->servers = array_map(function ($server) {
+            return $this->formatServer($server);
+        }, $servers);
+
+        // Set the connector options.
+        $this->retryLimit = (int) (isset($options['connector_retry_limit']) ? $options['connector_retry_limit'] : 20);
+        $this->retryWait = (int) (isset($options['connector_retry_wait']) ? $options['connector_retry_wait'] : 1000);
+
+        // Merge the global options shared by all Sentinel connections with
+        // connection-specific options
+        $clientOpts = array_merge($options, Arr::pull($servers, 'options', [ ]));
```

Comment:
In my opinion, there is no need to shorten variable names like this. But that is something the maintainer has to decide upon.

The `[ ]` at the end of the line does not need to have a space either.

Reply:
clientOpts is used as a variable name by the maintainer already. Same for the space inside [], that's done everywhere in the package. Not a fan of it either haha.